src/routes/extra_functionality_movie.py

```
@@ -0,0 +1,460 @@
+@router.post(
+    "/movies/{movie_id}/like",
+    response_model=MessageResponse,
+    summary="Like or dislike a movie",
+    description="Allows a user to like or dislike a movie. "
+    "If the user already has a reaction, it "
+    "will be updated or removed based on the new input.",
+)
+async def like_or_dislike_movie(
+    movie_id: int,
+    data: LikeDislikeSchema,
+    db: AsyncSession = Depends(get_db),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)